### DISCUSSION TITLE
LibreChat assumes that Redis is operating in Cluster mode; just hangs when trying to connect to a Standalone Redis server.

### DISCUSSION THREAD

**Discussion by cmurtaugh** [danny-avila/LibreChat#2844] [Feature Requests & Suggestions]
### What happened?

We've set the `USE_REDIS` and `REDIS_URI` variables to point to our Redis installation (an AWS ElastiCache cluster) which is operating in Standalone mode. If we don't enable social logins, LibreChat operates normally and the `keyv` module successfully caches data in our Redis server. However, if we enable social logins, LibreChat creates another Redis client for use by `connect-redis` for storing the user sessions. The [code that creates this client object](https://github.com/danny-avila/LibreChat/blob/8e66683577306fd9ebb94e39d1223f8baadb0943/api/cache/redis.js#L3) assumes that the server is operating in Cluster mode, and fails to connect (times out) if the server is not in Cluster mode. 

By contrast, `keyv` uses more sophisticated logic to create the correct Redis client for the server mode being used, so it works with both Standalone and Cluster modes.

One possible solution would be to add a `REDIS_MODE` env var or something similar so that we could specify whether our Redis server is Standalone or Cluster.

### Steps to Reproduce

1. Set up a Standalone Redis server.
2. Set the `USE_REDIS` and `REDIS_URI` env vars.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Redis integration is considered experimental, and no official configuration is recommended/tested.

I think both usages are quite separate in scope. I’ve thought about setting different URI env var for user sessions. But if using the same can work with the exact mode set, than I can look into this.
